src/main/java/codeu/controller/ResetServlet.java

```diff
@@ -50,27 +53,42 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
   public void doPost(HttpServletRequest request, HttpServletResponse response)
           throws IOException, ServletException {
     String username = request.getParameter("username");
+    // For next PR
     String password = request.getParameter("password");
 
-    if(password != null){
-      // ADD the emaillll
-    }
-
     if(username != null) {
+      User user = userStore.getUser(username);
       if (!userStore.isUserRegistered(username)) {
         request.setAttribute("error", "Username does not exist.");
         request.setAttribute("isReset", "false");
         request.getRequestDispatcher("/WEB-INF/view/reset.jsp").forward(request, response);
         return;
       }
-      if (userStore.getUser(username).getEmail().equals("")) {
+      if (user.getEmail().equals("")) {
         request.setAttribute("error", "No email. Sorry, we cannot retrieve the password.");
         request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
         return;
       }
-      request.setAttribute("username", username);
-      request.setAttribute("isReset", "true");
-      request.getRequestDispatcher("/WEB-INF/view/reset.jsp").forward(request, response);
+      // The case for sending a message
+      if(password == null) {
+        String url = request.getRequestURL().toString();
+        int last = url.lastIndexOf("/");
+        boolean sent = Util.sendEmail(username, user.getEmail(), user.getPasswordHash().substring(10), url.substring(0, last + 1) + "login");
+        if (sent) {
+          request.setAttribute("username", username);
+          request.setAttribute("isReset", "true");
+          request.setAttribute("sent", "The code was sent to: " + Util.transform(user.getEmail()));
+          request.getRequestDispatcher("/WEB-INF/view/login.jsp").forward(request, response);
+          return;
+        } else {
+          request.setAttribute("error", "Something went wrong with the sent. Create a new account");
+          request.getRequestDispatcher("/WEB-INF/view/register.jsp").forward(request, response);
+          return;
```

Comment:
Could the message for line 84 could be more clearer?

Reply:
It can, but it may require deciphering the type of error we are getting from the sendEmail() function. This can be complicated but I will look and see if it can be improve. Good point though!

Reply:
Ok cool! If the errors may vary, could we just say like "Something went wrong. Create a new account."? Using "Something went wrong with the sent" may be confusing